staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -901,6 +901,7 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 		}
 		resp, err := client.Do(req)
 		updateURLMetrics(ctx, r, resp, err)
+		metrics.RequestSize.Observe(ctx, r.verb, r.URL().Host, float64(req.ContentLength))
```

Comment:
is ContentLength always populated? is there ever a case we should ignore (-1 or 0 with non-nil body?)

Reply:
I honestly can't say, I was trying to skim the golang code but found it confusing, I will add an if to avoid the -1 and 0 corner cases to be safe

Reply:
the condition is weird, but is based on 
https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/net/http/request.go;l=919-927;drc=e6dda19888180c5159460486d30c0412e4980748

Reply:
> is ContentLength always populated? is there ever a case we should ignore (-1 or 0 with non-nil body?)

ignored on that case